src/util/trace.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_UTIL_TRACE_H
+#define BITCOIN_UTIL_TRACE_H
+
+#if /* !defined(ENABLE_TRACING) || */ defined(WIN32)
```

Comment:
`ENABLE_TRACING` is probably intended be provided a configure option (e.g. `--enable-ebpf`) later on?

Reply:
correct. I think having runtime probes would be neat to have in release builds, then you could tap into them in production. This shouldn't be a problem since traces are basically no-ops if there are no probes attached. This is what linux does of course. perhaps it would be a thing you could explicitly disable if you wanted to, or if configure couldn't find the sdt header.

Reply:
although it's not technically a no-op if the macro arg does some amount of work, like serializing an ip address to a string. so trace invocations shouldn't try to serialize anything or do work. you can always write parsers on the bpf side.

Reply:
> correct. I think having runtime probes would be neat to have in release builds,

Agree, I'm fine with enabling it by default when the prerequisites are installed,

> although it's not technically a no-op if the macro arg does some amount of work,

Yes, this is something to be careful with. The `LogPrintf`macro avoids this by evaluating its arguments only when debugging is enabled.
But I think in most cases it's not a problem, we'll only want to trace values that are readily available anyway.

Reply:
I've added a commit that detects sys/sdt.h and enables tracing if it finds it